cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/PhysicalStructureChange.java

```diff
@@ -157,8 +146,8 @@ protected void validateInternal(CwmsDTOValidator validator) {
         validator.validateRequiredFields(this);
         validator.validateCollection(getSettings());
     }
-
-    public static final class Builder<T extends Setting> {
+    
+    public abstract static class Builder<B extends Builder<B, T>, T extends Setting> {
```

Comment:
I can be a little annoying to type been can be really helpful when reading code to use a generic name that's distruptive. Like I have no idea what T is here until I go look at the TurbineChange class

Reply:
I usually use an otherwise normal variable name but then end it with "Type". Imagine reading this with https://github.com/opendcs/gherkin-to-kiwi/blob/main/lib/src/main/java/org/opendcs/testing/util/FailableResult.java with just S and F... or the way some people do it L and R.

Reply:
I'm not totally on board there. The Java convention explicitly call out the conventions for generic naming and refute the normal word name in order to not confuse with other classes or interfaces. https://docs.oracle.com/javase/tutorial/java/generics/types.html

I'd be even more confused looking for SettingType and not finding it to then realize it's a generic.

Reply:
I agree that the single character naming for java generics is the standard.  I'd be willing to give meaningful short names a try where longer names might make things clearer.   An alternative is to add documentation to the class declaration which explains what the generic holds.  Sometimes the generics don't make a lot of sense and they just have to be added in order to make things work.  In those cases I'm not sure any amount of naming or documentation would help.  In my experience anything above a class that takes a pair of generics wasn't designed from scratch but written to accommodate a specific use case.   When that is true a comment providing specific concrete examples can be as helpful as a long variable name.  Something like  ```// S - The type of the Settings class which corresponds to the Change. For example, For the TurbineChange class S would be TurbineSettings.```   Or something like that...

Reply:
1. I'm going to call oracle just totally full of shit here, we write code for others to read.
2. I'm asserting this for code we create, not filling in needs for other things.

But anyways. Javadoc *does* handle those parameters at the class level, you can see that in my example above. So we should at least put the meaningful description there.

Even C++ has gotten away from single character names, at least for anything that isn't blindingly obvious (granted with a c++ template it goes way beyond what a java generic, but the point stands, it's a *name*.

Reply:
Personally, I like the generic naming convention from C#.  It's basically a single character if it's self explanatory, otherwise you prefix your generic name with T like TSettingsType.
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#type-parameter-naming-guidelines
That probably works better for something like C++ or C# because they also prefer prefixes on class names like interfaces and abstractions.  So you're already in that environment of looking at the class name to identify what kind of thing it is.